torchao/float8/float8_linear_utils.py

```diff
@@ -303,13 +311,16 @@ def inner_func():
 
         # Calculate the new scales from the updated history stacks
         new_input_scales = amax_history_to_scale_stack(
-            fp8_input_amax_history_stack, e4m3_dtype, x_dtype, scale_fn_recipe
+            fp8_input_amax_history_stack, input_dtype, x_dtype, scale_fn_recipe
```

Comment:
will likely have to rebase on top of https://github.com/pytorch/ao/pull/1329 which changed this line